yargs.js

```diff
@@ -301,42 +301,58 @@ function Yargs (processArgs, cwd, parentRequire) {
     return self
   }
 
-  self.demand = self.required = self.require = function (keys, max, msg) {
+  self.demandOption = self.demand = self.required = self.require = function (keys, max, msg) {
```

Comment:
@maxrimue that is an interesting point. I feel like `demand` traditionally accepted `keys` as `number` only if it was a command, otherwise we would set it up as `demand: true` in option config. If it was a `string` we would dealing with an option. Please correct if I misunderstood that somehow. 

Which means `demand` is effectively the same as `demandOption`, because all I did is take out the `keys` as `number` logic and put it into `demandCommand`. Does that make sense?